core/pta/device.c

```diff
@@ -88,14 +88,21 @@ static TEE_Result invoke_command(void *pSessionContext __unused,
 {
 	TEE_Result res = TEE_SUCCESS;
 	uint32_t rflags = 0;
+	/*
+	 * FIXME: this should also be true if CFG_RPMB_ANNOUNCE_PROBE_CAP is
+	 * enabled but the kernel does not support OP-TEE RPMB operations
+	 * (probe failed)
+	 */
+	bool rpmb_needs_supp = !IS_ENABLED(CFG_RPMB_ANNOUNCE_PROBE_CAP);
 
 	switch (nCommandID) {
 	case PTA_CMD_GET_DEVICES:
 		rflags = TA_FLAG_DEVICE_ENUM;
 		break;
 	case PTA_CMD_GET_DEVICES_SUPP:
 		rflags = TA_FLAG_DEVICE_ENUM_SUPP;
-		if (IS_ENABLED(CFG_REE_FS))
+		if (IS_ENABLED(CFG_REE_FS) ||
+		    (IS_ENABLED(CFG_RPMB_FS) && rpmb_needs_supp))
 			rflags |= TA_FLAG_DEVICE_ENUM_TEE_STORAGE_PRIVATE;
```

Comment:
I wonder if `TA_FLAG_DEVICE_ENUM_TEE_STORAGE_PRIVATE` is sufficient.
If a TA uses the RPMB_FS and expects to be enumerated, it should set a `TA_FLAG_DEVICE_ENUM_TEE_STORAGE_RPMB`-like flag, regardless of whether the REE_FS is enabled or not.
That comment may be a bit outside the scope of this P-R.

Reply:
That's probably right, but the fix as it is at least allows the test mentioned in #7215 to pass. So maybe we should keep it like so for the moment.